### PR TITLE
Adding 403 Page to Dashboard

### PR DESCRIPTION
**Pull request by ccaprio97**
## What’s changing?

Adding 403 Page to Dashboard

## Why?

NAC-1837

## This PR has:

- [X] **JIRA ticket(s)** (_Must be listed in PR request and branch name should follow convention NAC-123-description_)<description>

- [ ] **Security Related Changes** (_Security Tab questions have been answered on JIRA ticket(s)_)

- [X] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Documented API changes** (_required on public classes and methods, and elsewhere as appropriate_)
    - [ ] External API documentation - OpenApi docs and Generated Docs (e.g. https://github.com/18F/piipan/tree/dev/docs/openapi) 
    - [ ] Internal API documentation (e.g. https://github.com/18F/piipan/tree/dev/metrics/docs/openapi)   

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)
    - [ ] Have IaC scripts run successfully against an existing deployed environment?
    - [ ] Have IaC scripts run successfully against a newly deployed environment?
    - [ ] Have any new bash scripts been given the appropriate permissions? (e.g. git update-index --chmod=+x $script)
    - [ ] Do the IaC changes require deleting existing resources?

## Anything else?

_Add other details that are helpful for review here_

## Code Review Checklist:

- [x] Automated checks all pass? (code builds successfully, tests pass, test coverage meets/exceeds threshold, etc.) 
- [x] Have all requirements and Acceptance Criteria been met?
- **If there are code changes**
    - [x] Is there no unnecessary complexity? Code is readable, maintainable, etc.
    - [x] Do the changes follow SOLID principles?
    - [x] Are the changes robust? Concurrency considered? Performance? Security (e.g. no sql injections)?
    - [ ] Are newly added dependencies needed and appropriate? Are dependency files (e.g. packages.json & packages.lock.json) updated?
    - [ ] For any new projects/solutions, have they been added to code coverage files?
    - [x] Public classes & methods are documented    
- **If there are API changes**
    - [ ] Does API make sense? Useful and concise?
    - [ ] Are existing API interfaces/contracts intact? i.e. no breaking changes
    - [ ] Are API tests included?
    - [ ] Are API responses appropriate and documented? (200s, 201, 404s, etc.)
    - [ ] Are internals kept private and not leaked into the API?
    - [ ] External API documentation - OpenApi docs and Generated Docs 
    - [ ] Internal API documentation    
- **If there are IAC changes**
    - [ ] IAC Scripts are readable & maintainable
    - [ ] IAC Scripts are documented
    - [ ] IAC scripts pass required checks
    - [ ] IAC scripts were run successfully by author
- **General Logging Questions**
    - [ ] Do new capabilities (or changed capabilities) include logging?
    - [ ] Is PII not logged?
- **General Error Handling Questions**
    - [ ] Has exception handling been added for new capabilities (or changed capabilities)?
    - [ ] Are errors/exceptions properly communicated to end-users? Were there requirements for user facing error messages?
    - [ ] Are errors/exceptions logged?
    - [ ] Is PII kept out of error/exception logging?
- **If there are Developer Documentation Changes**
    - [ ] General Project documentation (under piipan\docs. e.g. iac.md, architecture.md, etc.)
    - [ ] Github documentation additions/changes (Readmes)
- **General automated test questions**
    - [ ] End-to-End Tests provided (if code implements new user story)
    - [x] Are tests sufficient?
    - [x] Corner cases considered?
    - [ ] Are performance tests needed?

### CODE REVIEW

File: dashboard/tests/Piipan.Dashboard.Tests/ServerErrorPageTests.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Piipan.Dashboard.Pages;
+using Xunit;
+
+namespace Piipan.Dashboard.Tests
+{
+    /// <summary>
+    /// Tests for the Server Error page. This is the page that gets shown when the user gets any 500 error.
+    /// </summary>
+    public class ServerErrorPageTests : BasePageTest
```
Comment (dborgen2) [inline dashboard/tests/Piipan.Dashboard.Tests/ServerErrorPageTests.cs:11]: Can you add tests for NotAuthorized similar to these?

File: dashboard/src/Piipan.Dashboard/Pages/NotAuthorized.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Piipan.Dashboard.Pages
+{
+    [AllowAnonymous]
+    public class NotAuthorizedModel : PageModel
```
Comment (dborgen2) [inline dashboard/src/Piipan.Dashboard/Pages/NotAuthorized.cshtml.cs:12]: Looks like we need a comment here for this public class and public class methods. If the QueryTool version doesn't have them, if you wouldn't mind copy/pasting them to the QueryTool too that would be great.

**Review by dborgen2 — APPROVED** [review]
LGTM
